storm-client/src/jvm/org/apache/storm/cluster/StormClusterStateImpl.java

```diff
@@ -423,32 +424,35 @@ public void supervisorHeartbeat(String supervisorId, SupervisorInfo info) {
     }
 
     /**
-     * if znode exists and to be not on?, delete; if exists and on?, do nothing; if not exists and to be on?, create; if not exists and not on?, do nothing;
+     * if znode exists and timestamp is 0?, delete; if exists and timestamp is larger than 0?, do nothing;
+     * if not exists and timestamp is larger than 0?, create the node and set the timestamp; if not exists and timestamp is 0?, do nothing;
      * 
      * @param stormId
      * @param node
      * @param port
-     * @param on
+     * @param timestamp
```

Comment:
It might be nicer to split this method into a create and delete variant, but we should at least put in the javadoc that timestamp 0 means "turn off"